lib/router/transition.js

```diff
@@ -235,7 +235,18 @@ Transition.prototype = {
     // TODO: add tests for merged state retry()s
     this.abort();
     var newTransition = this.router.transitionByIntent(this.intent, false);
-    newTransition.method(this.urlMethod);
+
+    // inheriting a `null` urlMethod is not valid
+    // the urlMethod is only set to `null` when
+    // the transition is initiated *after* the url
+    // has been updated (i.e. `router.handleURL`)
+    //
+    // in that scenario, the url method cannot be
+    // inherited for a new transition because then
+    // the url would not update even though it should
+    if (this.urlMethod !== null) {
+      newTransition.method(this.urlMethod);
```

Comment:
I'm a bit confused actually. I thought that the `urlMethod` could be either `update` or `replace`. Under what circumstances is it `null`?